digdag-core/src/main/java/io/digdag/core/database/DatabaseTaskQueueFactory.java

```diff
@@ -1,187 +1,40 @@
 package io.digdag.core.database;
 
-import java.util.List;
-import java.io.IOException;
 import com.google.inject.Inject;
-import com.google.common.base.Throwables;
-import com.google.common.collect.ImmutableList;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import io.digdag.core.repository.ResourceConflictException;
-import io.digdag.core.repository.ResourceNotFoundException;
 import io.digdag.client.config.Config;
-import io.digdag.spi.TaskQueue;
 import io.digdag.spi.TaskQueueFactory;
 import io.digdag.spi.TaskQueueServer;
 import io.digdag.spi.TaskQueueClient;
-import io.digdag.spi.TaskStateException;
 import io.digdag.spi.TaskRequest;
-import io.digdag.core.database.DatabaseTaskQueueStore.LockResult;
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import static io.digdag.core.queue.QueueSettingStore.DEFAULT_QUEUE_NAME;
 
 public class DatabaseTaskQueueFactory
     implements TaskQueueFactory
 {
-    private final DatabaseTaskQueueStore store;
-    private final ObjectMapper mapper;
+    private final DatabaseTaskQueueServer database;
     private final Object sharedTaskSleepHelper = new Object();
     private final Object taskSleepHelper = new Object();
 
     @Inject
-    public DatabaseTaskQueueFactory(DatabaseTaskQueueStore store, ObjectMapper mapper)
+    public DatabaseTaskQueueFactory(DatabaseTaskQueueServer database)
     {
-        this.store = store;
-        this.mapper = mapper;
+        this.database = database;
     }
 
+    @Override
     public String getType()
     {
         return "database";
```

Comment:
So should we change this line https://github.com/treasure-data/digdag/blob/c1277203d608ef2ce9919f31cec0821751fabb8b/digdag-core/src/main/java/io/digdag/core/queue/TaskQueueManager.java#L33 to use this method?